### PR TITLE
Backend Finish Sign Up event Statsig

### PR DESCRIPTION
**Pull request by hannahbergam**
This PR adds a new event to the back end for 'finish_sign_up' that will more accurately represent when a new user has been created. To avoid having to run a database query to pull in the nces_id to the user reporting, I'm sending a boolean for has_school in the metadata instead. If this is true, there is a school_info row for this user.

Screenshot 2024-04-29 at 10 39 23 AM

## Warning!!

The [AP CSP Create Performance Task](https://apcentral.collegeboard.org/courses/ap-computer-science-principles/exam) is in progress. The most critical dates are from April 3 - April 30, 2024. Please consider any risk introduced by this PR that could affect our students taking AP CSP. Code.org students taking AP CSP primarily use App Lab for their Create Task, however a small percent use Game Lab. Carefully consider whether your change has any risk of alterering, changing, or breaking anything in these two labs. Even small changes, such as a different button color, are considered significant during this time period. Reach out to the Student Learning team or Curriculum team for more details.

## Links

- spec: []()
- jira ticket: https://codedotorg.atlassian.net/browse/ACQ-1607

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/controllers/registrations_controller.rb [hunk 1]
```diff
@@ -135,6 +135,15 @@ def create
           metadata: metadata,
         )
       end
+      has_school = current_user.school_info_id?
+      event_metadata = {
+        'has_school' => has_school,
+      }
+      Metrics::Events.log_event(
```
Comment (nicklathe) [inline dashboard/app/controllers/registrations_controller.rb:148]: I'm fine with having two different events here, but _if_ we wanted to combine them, what if we did something like this?

```
# initialize empty variables
metadata, event_name = nil

if Policies::Lti.lti? current_user
  # build up LTI metadata
  metadata = {
    'lms_name' => lms_name,
  }
  # set LTI event_name
  event_name = 'lti_user_created'
} else {
  # build up non-LTI metadata
  event_metadata = {
    'has_school' => current_user.school_info_id?
  }
  # set non-LTI event_name
  event_name = 'Sign Up Finished Backend'
}

# Send event with whatever values we built
Metrics::Events.log_event(
  user: current_user,
  event_name: event_name
  metadata: metadata,
)
```

Comment (nicklathe) [reply dashboard/app/controllers/registrations_controller.rb:148]: Although I'm realizing now that the default/global 'Sign Up Finished Backend' is probably still desirable for LTI users, so this is probably not what we'd want to do.

**Review by nicklathe — APPROVED** [review]
I left a comment, but realized it's probably not a good suggestion. From an event logging perspective, this LGTM.

**Review by bethanyaconnor — APPROVED** [review]
Change looks good! Do you know how we'll know what experiment group the user is in?

**Comment by hannahbergam** [comment]
> Change looks good! Do you know how we'll know what experiment group the user is in?

Yes- this is what I was working on with @nicklathe last week! Nick- that will be part of the user metadata right?

**Comment by hannahbergam** [comment]
> > Change looks good! Do you know how we'll know what experiment group the user is in?
> 
> Yes- this is what I was working on with @nicklathe last week! Nick- that will be part of the user metadata right?

@nicklathe Or am I now remembering that you are defaulting to *not sending that data, and I'll have to explicitly ask for it somehow...

**Comment by nicklathe** [comment]
> > > Change looks good! Do you know how we'll know what experiment group the user is in?
> > 
> > 
> > Yes- this is what I was working on with @nicklathe last week! Nick- that will be part of the user metadata right?
> 
> @nicklathe Or am I now remembering that you are defaulting to *not sending that data, and I'll have to explicitly ask for it somehow...

Sorry, I should have mentioned this in my review. You'll need to add this argument to the `Metrics::Events.log_event` method: `get_enabled_experiments: true,` as outlined in [this PR](https://github.com/code-dot-org/code-dot-org/pull/58261). That will populate a list of enabled experiment names, under the `custom_ids` key in the user object.

**Comment by hannahbergam** [comment]
> > > > Change looks good! Do you know how we'll know what experiment group the user is in?
> > > 
> > > 
> > > Yes- this is what I was working on with @nicklathe last week! Nick- that will be part of the user metadata right?
> > 
> > 
> > @nicklathe Or am I now remembering that you are defaulting to *not sending that data, and I'll have to explicitly ask for it somehow...
> 
> Sorry, I should have mentioned this in my review. You'll need to add this argument to the `Metrics::Events.log_event` method: `get_enabled_experiments: true,` as outlined in [this PR](https://github.com/code-dot-org/code-dot-org/pull/58261). That will populate a list of enabled experiment names, under the `custom_ids` key in the user object.

Updated! Re-requested review just in case.

**Review by nicklathe — APPROVED** [review]
Looks good!

**Pull request merged**